tests/posix/xsi_threads_ext/src/main.c

```diff
@@ -0,0 +1,282 @@
+}
+
+static inline void can_create_thread(const pthread_attr_t *attrp)
+{
+	create_thread_common(attrp, true, true);
+}
+
+ZTEST(xsi_threads_ext, test_pthread_attr_getstack)
+{
+	void *stackaddr = (void *)BIOS_FOOD;
+	size_t stacksize = BIOS_FOOD;
+
+	/* degenerate cases */
+	{
+		if (false) {
+			/* undefined behaviour */
+			zassert_equal(pthread_attr_getstack(NULL, NULL, NULL), EINVAL);
+			zassert_equal(pthread_attr_getstack(NULL, NULL, &stacksize), EINVAL);
+			zassert_equal(pthread_attr_getstack(NULL, &stackaddr, NULL), EINVAL);
+			zassert_equal(pthread_attr_getstack(NULL, &stackaddr, &stacksize), EINVAL);
+			zassert_equal(pthread_attr_getstack(&uninit_attr, &stackaddr, &stacksize),
+				      EINVAL);
+		}
+		zassert_equal(pthread_attr_getstack(&attr, NULL, NULL), EINVAL);
+		zassert_equal(pthread_attr_getstack(&attr, NULL, &stacksize), EINVAL);
+		zassert_equal(pthread_attr_getstack(&attr, &stackaddr, NULL), EINVAL);
+	}
+
+	zassert_ok(pthread_attr_getstack(&attr, &stackaddr, &stacksize));
+	zassert_not_equal(stackaddr, (void *)BIOS_FOOD);
+	zassert_not_equal(stacksize, BIOS_FOOD);
+}
+
+ZTEST(xsi_threads_ext, test_pthread_attr_setstack)
+{
+	void *stackaddr;
+	size_t stacksize;
+	void *new_stackaddr;
+	size_t new_stacksize;
+
+	/* valid values */
+	zassert_ok(pthread_attr_getstack(&attr, &stackaddr, &stacksize));
+
+	/* degenerate cases */
+	{
+		if (false) {
+			/* undefined behaviour */
+			zassert_equal(pthread_attr_setstack(NULL, NULL, 0), EACCES);
+			zassert_equal(pthread_attr_setstack(NULL, NULL, stacksize), EINVAL);
+			zassert_equal(pthread_attr_setstack(NULL, stackaddr, 0), EINVAL);
+			zassert_equal(pthread_attr_setstack(NULL, stackaddr, stacksize), EINVAL);
+			zassert_equal(pthread_attr_setstack((pthread_attr_t *)&uninit_attr,
+							    stackaddr, stacksize),
+				      EINVAL);
+		}
+		zassert_equal(pthread_attr_setstack(&attr, NULL, 0), EACCES);
+		zassert_equal(pthread_attr_setstack(&attr, NULL, stacksize), EACCES);
+		zassert_equal(pthread_attr_setstack(&attr, stackaddr, 0), EINVAL);
+	}
+
+	/* ensure we can create and join a thread with the default attrs */
+	can_create_thread(&attr);
+
+	/* set stack / addr to the current values of stack / addr */
+	zassert_ok(pthread_attr_setstack(&attr, stackaddr, stacksize));
+	can_create_thread(&attr);
+
+	/* qemu_x86 seems to be unable to set thread stacks to be anything less than 4096 */
+	if (!IS_ENABLED(CONFIG_X86)) {
+		/*
+		 * check we can set a smaller stacksize
+		 * should not require dynamic reallocation
+		 * size may get rounded up to some alignment internally
+		 */
+		zassert_ok(pthread_attr_setstack(&attr, stackaddr, stacksize - 1));
+		/* ensure we read back the same values as we specified */
+		zassert_ok(pthread_attr_getstack(&attr, &new_stackaddr, &new_stacksize));
+		zassert_equal(new_stackaddr, stackaddr);
+		zassert_equal(new_stacksize, stacksize - 1);
+		can_create_thread(&attr);
+	}
+
+	if (IS_ENABLED(DYNAMIC_THREAD_ALLOC)) {
```

Comment:
```suggestion
	if (IS_ENABLED(CONFIG_DYNAMIC_THREAD_ALLOC)) {
```

Where is `CONFIG_DYNAMIC_THREAD_ALLOC` set in the `common` test suite and why doesn't it appear in this test suite?